pyhive/hive.py

```diff
@@ -337,18 +346,14 @@ def fetch_logs(self):
 
 def _unwrap_column(col):
     """Return a list of raw values from a TColumn instance."""
-    for attr, wrapper in col.__dict__.iteritems():
+    for attr, wrapper in iteritems(col.__dict__):
         if wrapper is not None:
-            values = wrapper.values
+            result = wrapper.values
             nulls = wrapper.nulls  # bit set describing what's null
-            assert isinstance(nulls, str)
-            if attr == 'stringVal':
-                result = [val.decode('utf-8') for val in values]
-            else:
-                result = values
+            assert isinstance(nulls, (str, bytes))
```

Comment:
apparently `bytes is str => true`, so this can probably be `isinstance(nulls, bytes)`